packages/ejson/lib/src/encoding.dart

```diff
@@ -0,0 +1,223 @@
+
+EJsonValue _encodeUuid(Uuid uuid) => _encodeBinary(uuid.bytes.asUint8List(), "04");
+
+EJsonValue _encodeBinary(Uint8List buffer, String subtype) => {
+      '\$binary': {
+        'base64': base64.encode(buffer),
+        'subType': subtype,
+      },
+    };
+
+EJsonValue _encodeObjectId(ObjectId objectId) => {'\$oid': objectId.hexString};
+
+/// Exception thrown when no encoder is registered for the type of a [value].
+class MissingEncoder implements Exception {
+  final Object value;
+
+  MissingEncoder(this.value);
+
+  @override
+  String toString() => 'Missing encoder for type ${value.runtimeType} ($value)';
+}
+
+extension BoolEJsonEncoderExtension on bool {
+  /// Converts this [bool] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeBool(this);
+}
+
+extension DateTimeEJsonEncoderExtension on DateTime {
+  /// Converts this [DateTime] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeDate(this);
+}
+
+extension DefinedEJsonEncoderExtension on Defined<dynamic> {
+  /// Converts this [Defined] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeDefined(this);
+}
+
+extension DoubleEJsonEncoderExtension on double {
+  /// Converts this [double] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeDouble(this);
+}
+
+extension IntEJsonEncoderExtension on int {
+  /// Converts this [int] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson({bool long = true}) => _encodeInt(this, long: long);
+}
+
+extension KeyEJsonEncoderExtension on Key {
+  /// Converts this [Key] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeKey(this);
+}
+
+extension ListEJsonEncoderExtension on List<Object?> {
+  /// Converts this [List] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeArray(this);
+}
+
+extension MapEJsonEncoderExtension on Map<dynamic, dynamic> {
+  /// Converts this [Map] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeDocument(this);
+}
+
+extension NullEJsonEncoderExtension on Null {
+  /// Converts this [Null] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => null;
+}
+
+extension NullableObjectEJsonEncoderExtension on Object? {
+  /// Converts this [Object] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeAny(this);
+}
+
+extension ObjectIdEJsonEncoderExtension on ObjectId {
+  /// Converts this [ObjectId] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeObjectId(this);
+}
+
+extension StringEJsonEncoderExtension on String {
+  /// Converts this [String] to EJson
+  @pragma('vm:prefer-inline')
+  EJsonValue toEJson() => _encodeString(this);
+}
+
+extension SymbolEJsonEncoderExtension on Symbol {
+  /// Extract the name of this [Symbol]
+  String get name {
+    final full = toString();
+    return full.substring(8, full.length - 2);
```

Comment:
Can we add a comment that explains these magic numbers?

Reply:
Done